docker/secondary-analysis-python/tools/submit/submit/submit.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+
+import requests
+import json
+import argparse
+
+def run(submit_url, analysis_json_path):
+    print('Starting submission with url: {0}'.format(submit_url))
+
+    # 1. Get envelope url
+    response = requests.get(submit_url)
+    envelope_url = get_entity_url(response.json(), 'submissionEnvelopes')
+
+    # 2. Create envelope, get analysis and submission urls
+    print('Creating submission envelope at {0}'.format(envelope_url))
+    response = requests.post(envelope_url, '{}')
+    envelope_js = response.json()
+    analyses_url = get_entity_url(envelope_js, 'analyses')
+    submission_url = get_entity_url(envelope_js, 'submissionEnvelope')
+
+    # 3. Create analysis, get input bundles url, file refs url
+    print('Creating analysis at {0}'.format(analyses_url))
+    json_header = {'Content-type': 'application/json'}
+    with open(analysis_json_path) as f:
+        analysis_json_contents = json.load(f)
+    response = requests.post(analyses_url, headers = json_header, data = analysis_json_contents)
+    print(response.text)
+    analysis_js = response.json()
+    input_bundles_url = get_entity_url(analysis_js, 'add-input-bundles')
+    file_refs_url = get_entity_url(analysis_js, 'add-file-reference')
+
+    # 4. Add input bundles
+    print('Adding input bundles at {0}'.format(input_bundles_url))
+    input_bundle_uuid = get_input_bundle_uuid(analysis_json_contents)
+    response = requests.post(input_bundles_url, data = {"bundleUuids": [input_bundle_uuid]})
+    print(response.text)
+
+    # 5. Add file references
+    print('Adding file references at {0}'.format(file_refs_url))
+    output_files = get_output_files(analysis_json_contents)
+    response = requests.put(file_refs_url, headers = json_header, data = output_files)
+    print(response.text)
+
+    # 6. Confirm submission
+    confirmation_url = '{0}/{1}'.format(submission_url, 'confirmation')
+    print('Confirming submission at {0}'.format(confirmation_url))
+    response = requests.put(confirmation_url, headers = json_header)
+    print(response)
+
```

Comment:
I think these calls could be a cool place to define a `notify` decorator, if someone wants to learn how those work at some point (low priority). It could look like this: 

```
from functools import wraps

def notify(notification_string):
    def decorator(func):
        @wraps(func)
        def wrapper(*args, **kwargs):
            print(notification_string)
            response = func(*args, **kwargs)
            print(response)
            return response
        return wrapper
    return decorator
```
Then, you can use this as follows: 

```
@notify('my_message_here')
def get_input_bundle_uuid(analysis_json):
    bundle = analysis_json['input_bundles'][0]
    uuid = bundle.split('/')[-2]
    print('Input bundle uuid {0}'.format(uuid))
    return uuid         
```
then you just invoke it with: 
```
input_bundle_uuid = get_input_bundle_uuid(analysis_json_contents)
```
and all the printing is already done. 

I doubt the code is runnable as I've coded it, but the idea is that you could abstract the notification and make it easy to write this. Flask may also already have this. :)

Reply:
To make clear how decorators work: https://realpython.com/blog/python/primer-on-python-decorators/

Reply:
I don't fully understand how this works, so would like to talk to you more about it.

Reply:
Probably true that it is overkill. It's a cool concept worth knowing about though. Happy to chat about it.